### ISSUE TITLE
Discovery rules already exits.

### ISSUE THREAD

**Issue by fallboyz** [sergiotocalini/habbixy#1]
Hi.
I have been using habbixy from a long time ago.
This time a new version of haproxy comes out, and the existing server was removed, a new server was configured, and the template was installed again.

centos7
haproxy 2.0.1

Can not create item: item with the same key "habbixy [stat, {#NAME}, FRONTEND, bin]" already exists.
Can not create item: item with the same key "habbixy [stat, {#NAME}, FRONTEND, bout]" already exists.
Can not create item: item with same key "habbixy [stat, {#NAME}, FRONTEND, scur]" already exists.
Can not create item: item with same key "habbixy [stat, {#NAME}, FRONTEND, slim]" already exists.
Can not create item: item with same key "habbixy [stat, {#NAME}, FRONTEND, status]" already exists.

Can not accurately apply filter: no value received for macro "{#BACKEND_NAME}".

Discovery is not coming out with this message. Do you know why?
Other data comes in well.
I'm sorry I did not translate it into Google Translator because I can not speak English.

**Comment by sergiotocalini** [comment]
Hi @fallboyz 
It may happens because your frontend are repeated or are starting with the same pattern and have spaces.
Place do a git pull I introduced a change on the zabbix_agentd userparameters than maybe it can help you.
Let me know.
Best,
Sergio

**Comment by fallboyz** [comment]
Hi @sergiotocalini 

I am using haproxy 2.0 and using the master-worker option.
Previously I was using haproxy 1.8, and then I was using the master-worker option, but this didn't happen.

Perhaps my guess is that your script doesn't work with haproxy 2.0 and above.

After testing this and that, I noticed that with the socket plugged in it is no longer show stat or show info.

The result is that if you go to the next level with @master or @pid and show stat or show info, the information comes out.

So while I was wondering whether to solve it from the command line, I could see that @1 solved it.

In other words, on line 62 of your script,
echo "show $ {type}" | sudo socat $ {HAPROXY_SOCKET} stdio 2> / dev / null> $ {file}
In the part called
echo "@1 show $ {type}" | sudo socat $ {HAPROXY_SOCKET} stdio 2> / dev / null> $ {file}
To make your script run normally.

If in doubt, I recommend you test it.

I am using CentOS7 correctly, using HAProxy 2.0.1, and using the master-worker option.

Please test and reflect on my small contribution to your scripts.

Again, I'm not good at English, so I'm writing this article using Google Translator.
